### ISSUE TITLE
Bug: [no-unsafe-argument] Check unsafe arguments in Tagged templates

### ISSUE THREAD

**Issue by yeonjuan** [typescript-eslint/typescript-eslint#8745]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unsafe-argument

### Description

`no-unsafe-argument` does not report errors when unsafe arguments used in Tagged templates.

### Fail

```ts
function foo<T = number>(templates: TemplateStringsArray, arg: T) {}
declare const arg: any;
foo<number>`${arg}`;
```

### Pass

```ts
function foo<T = number>(templates: TemplateStringsArray, arg: T) {}
declare const arg: number;
foo<number>`${arg}`;
```

**Comment by JoshuaKGoldberg** [comment]
I'd say this is a bug. Tagged templates are fancy syntax for calling a function - we just tend to forget about them.
